### PR TITLE
Add stub onednn jit optimizer extension

### PR DESCRIPTION
**Pull request by StrongerXi**
Depends on #1038

### Summary
Add a do-nothing OneDNN JitOptimizerExtension.

OneDNN-specific JIT optimization passes will be added here in the future.

### Test Plan (required)
CI

### CODE REVIEW

File: flashlight/fl/tensor/backend/jit/opt/backends/onednn/OneDnnJitOptimizerExtension.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace fl {
+
+/**
+ * JIT graph optimziation specific to OneDNN backend, e.g., node fusion to
+ * leverage OneDNN post-ops.
+ */
+class OneDnnJitOptimizerExtension : public JitOptimizerExtension {
+ public:
+  static bool registered;
```
Comment (jacobkahn) [inline flashlight/fl/tensor/backend/jit/opt/backends/onednn/OneDnnJitOptimizerExtension.h:20]: This isn't needed anymore -- got rid of it in https://github.com/flashlight/flashlight/pull/1015

**Review by jacobkahn — APPROVED** [review]
LGTM, one quick nit

File: flashlight/fl/tensor/backend/jit/opt/backends/onednn/OneDnnJitOptimizerExtension.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "flashlight/fl/tensor/backend/jit/opt/backends/onednn/OneDnnJitOptimizerExtension.h"
+
+#include "flashlight/fl/tensor/backend/onednn/OneDnnBackend.h"
```
[the remark this answers is not in the document]
Comment (StrongerXi) [reply flashlight/fl/tensor/backend/jit/opt/backends/onednn/OneDnnJitOptimizerExtension.cpp:10]: Yeah so I thought of this `JitOptimizerExtension` as similar to `AutogradExtension`, which is defined closer to autograd.

**Comment by StrongerXi** [comment]
Rebase & address feedback.

**Pull request closed without merging**